packages/runtime/src/account.ts

```diff
@@ -24,6 +24,20 @@ const StateMap = "key-value";
 const globalState = "global-state";
 const localStateSchema = "local-state-schema";
 
+function checkAndSetAssetFields (fields: AssetModFields, asset: AssetDef): void {
+  if (!(fields.manager !== "" && asset.manager === "") && fields.manager !== undefined) asset.manager = fields.manager;
+  else if (fields.manager !== undefined) throw new RuntimeError(RUNTIME_ERRORS.ASA.BLANK_ADDRESS_ERROR);
+
+  if (!(fields.reserve !== "" && asset.reserve === "") && fields.reserve !== undefined) asset.reserve = fields.reserve;
+  else if (fields.reserve !== undefined) throw new RuntimeError(RUNTIME_ERRORS.ASA.BLANK_ADDRESS_ERROR);
+
+  if (!(fields.freeze !== "" && asset.freeze === "") && fields.freeze !== undefined) asset.freeze = fields.freeze;
+  else if (fields.freeze !== undefined) throw new RuntimeError(RUNTIME_ERRORS.ASA.BLANK_ADDRESS_ERROR);
+
+  if (!(fields.clawback !== "" && asset.clawback === "") && fields.clawback !== undefined) asset.clawback = fields.clawback;
+  else if (fields.clawback !== undefined) throw new RuntimeError(RUNTIME_ERRORS.ASA.BLANK_ADDRESS_ERROR);
+}
```

Comment:
maybe we can use loop?
```
for (x of ['manager', 'reserve', 'freeze', 'clawback'])
  if ....
```

Reply:
Sure.

Reply:
Also, let's the conditions:
```
if (fields.clawback !== "" && fields.clawback !== undefined && asset.clawback === "") throw ...
else if (fields.clawback !== "") assets.clawback = fields.clawback
```

Reply:
The `!` operator is just for the conditions `(fields.clawback !== "" && asset.clawback === "")` and not the whole line. Also, the loop implementation isn't working.

Reply:
What's the condition to throw an error? I think it's the one I put above.

Reply:
This implementation fails when `fields.clawback` is undefined.

Reply:
So, maybe we can remove the first component in the if: `fields.clawback !== ""`. What do you think?
I prefer to firstly have a condition for throw, and then to have a condition to update.

Reply:
can we use the loop instead of copy-pasting the `if-else`

Reply:
@man-of-code could you check if this works:
```js
for (const x of ['manager', 'reserve', 'freeze', 'clawback']) {
  const asaField = fields[x as keyof AssetModFields];
  if (isSetOrDefined(asaField) && !isSetOrDefined(asaField)) {
    throw new RuntimeError(RUNTIME_ERRORS.ASA.BLANK_ADDRESS_ERROR);
  } else if (asaField !== undefined && isSetOrDefined(asaField)) {
    asset[x as keyof AssetModFields] = asaField;
  }
}
```